scripts/submit.ts

```diff
@@ -14,14 +14,18 @@ async function simulateTransaction(transaction: BytesLike) {
 	const nonce = await provider.getTransactionCount(call.from as string)
 	console.log({ nonce })
 	if (call.nonce !== nonce) {
-		throw new Error('nonce must be 0')
+		throw new Error("nonce must be 0")
 	}
 	for (const property of ["hash", "type", "v", "r", "s"] as const) {
 		delete call[property]
 	}
 	console.log({ call })
 	const deployedBytecode = await provider.call(call as any)
-	console.log({ deployedBytecode })
+	const codeHash = ethers.utils.keccak256(ethers.utils.arrayify(deployedBytecode));
+	console.log({ deployedBytecode, codeHash })
+	if (codeHash != "0x2fa86add0aed31f33a762c9d88e807c475bd51d0f52bd0955754b2608f7e4989") {
```

Comment:
I would extract this codehash as a constant so it's not buried inside the code and can be used for other purposes

Reply:
We could share it in verify as well if we extract it: https://github.com/safe-global/safe-singleton-factory/blob/c3ea1239355f4e8b6ce13056284a6b4a4b6abb6d/scripts/verify.ts#L10